coderd/rbac/acl/updatevalidator.go

```diff
@@ -0,0 +1,108 @@
+package acl
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/uuid"
+
+	"github.com/coder/coder/v2/coderd/database"
+	"github.com/coder/coder/v2/coderd/database/dbauthz"
+	"github.com/coder/coder/v2/codersdk"
+)
+
+type UpdateValidator[Role codersdk.WorkspaceRole | codersdk.TemplateRole] interface {
+	Users() (map[string]Role, string)
+	Groups() (map[string]Role, string)
+	ValidateRole(role Role) error
```

Comment:
Can we comment what these method signatures are? The second string is not obvious by the interface alone. It's for error reporting